### ISSUE TITLE
Extract task does not find proto of library packaged with bootJar 

### ISSUE THREAD

**Issue by YannPerthuis** [google/protobuf-gradle-plugin#597]
Hello, 

I really appreciate the help as I've been stuck on this for a few days.

## Context

I have a spring boot application named foo, which contains protobuf files, and I package it with the spring "bootJar" task and publish it locally with maven-publish.

I have another bar application that uses foo as a dependency only for its protobuf files:

```groovy
dependencies {
    protobuf 'com.mycompany:foo:0.0.1-SNAPSHOT'
}
```

## Bug

The protobuf gradle plugin is not able to find the proto files of foo which are located in the BOOT-INF folder which is at the root of the jar.

## Workaround

It works when I package foo with the classic "jar" task and the proto files are directly in the root of the jar, but it doesn't work as I want

## Environment

Gradle 7.3.3
Java 17
Protobuf gradle plugin 0.8.19
Protoc artifact 3.20.1

Has anyone encountered this bug before ?

**Comment by YannPerthuis** [comment]
## Update

So, I realized that the foo package was not well pushed locally, it's done.


## Still a bug with bootjar task

Now the bar package extracts the foo proto files, but the problem is that they are all located in the BOOT-INF/classes folder in the foo jar and are extracted that way. This is the normal behavior when using the spring boot plugin for package app I agree, but the foo paths become wrong and depend on the spring framework. Moreover it generates me the following error due to the fact that the foo dep other bar dependencies embed the google, grpc and protoc-gen proto files :

```groovy
Execution failed for task ':generateProto'.
> protoc: stdout: . stderr: BOOT-INF/classes/google/api/billing.proto:19:1: warning: Import google/api/metric.proto is unused.
  BOOT-INF/classes/google/api/http.proto:33:21: "google.api.Http.rules" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:41:8: "google.api.Http.fully_decode_reserved_expansion" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:29:9: "google.api.Http" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto: "google.api.HttpRule.pattern" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:317:10: "google.api.HttpRule.selector" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:325:12: "google.api.HttpRule.get" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:328:12: "google.api.HttpRule.put" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:331:12: "google.api.HttpRule.post" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:334:12: "google.api.HttpRule.delete" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:337:12: "google.api.HttpRule.patch" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:343:23: "google.api.HttpRule.custom" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:352:10: "google.api.HttpRule.body" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:360:10: "google.api.HttpRule.response_body" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:365:21: "google.api.HttpRule.additional_bindings" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:313:9: "google.api.HttpRule" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:371:10: "google.api.CustomHttpPattern.kind" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:374:10: "google.api.CustomHttpPattern.path" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:369:9: "google.api.CustomHttpPattern" is already defined in file "google/api/http.proto".
  BOOT-INF/classes/google/api/http.proto:33:12: "google.api.HttpRule" seems to be defined in "google/api/http.proto", which is not imported by "BOOT-INF/classes/google/api/http.proto".  To use it here, please add the necessary import.
  BOOT-INF/classes/google/api/http.proto:343:5: "google.api.CustomHttpPattern" seems to be defined in "google/api/http.proto", which is not imported by "BOOT-INF/classes/google/api/http.proto".  To use it here, please add the necessary import.
  BOOT-INF/classes/google/api/http.proto:365:12: "google.api.HttpRule" seems to be defined in "google/api/http.proto", which is not imported by "BOOT-INF/classes/google/api/http.proto".  To use it here, please add the necessary import.
```

## Question

According to you, should bar adapt to all its dependencies (e.g. exclude from its dependencies all proto files like google's or grpc's in the BOOT-INF/classes folder), or should it be up to each of its dependencies to adopt the same structure (e.g. always put the proto files at the root of the jar)?

Depending on your answer would you have any idea how to do it ?

**Comment by YannPerthuis** [comment]
Finally in the meantime I found a workaround to put the protobuf files at the root of the jar created by the bootjar task in `build.gradle` file :

```groovy
bootJar {
    archiveClassifier = ''
    mainClass
    eachFile { details ->
        String packagedProtoPathPrefix = 'BOOT-INF/classes/'
        if (details.path.startsWith(packagedProtoPathPrefix) && details.path.endsWith('.proto')) {
            details.path = details.path.substring(packagedProtoPathPrefix.length())
        }
    }
}
```

Anyway, after further research (e.g. [here](https://www.baeldung.com/spring-boot-dependency)) it seems that it is not advisable to use the jar created by the bootJar task, but to create a second one to use it as a dependency.

Indeed the way spring packages the app is specific and can't be used as a dependency; unfortunately I don't think I saw this mentioned in the spring-boot-gradle-plugin documentation to raise any ambiguity.

I close the issue, thanks
